parsers/EIA.py

```diff
@@ -501,9 +522,9 @@ def fetch_production_mix(
         # As null values can cause problems in the estimation models if there's
```

Comment:
Oh man I thought this was fixed and removed already 👀

Looks like we have some cleanup to do (In another PR)

Reply:
yeah feels like parsers could use a bit of a make over - I'll write up a pitch for it

Reply:
This check is already included in the parser classes as far as I know, but we should have a snapshot test for all production parsers then we can make these changes with more confidence.